src/components/updated-terms-conds/updated-terms-conds-controller.ts

```diff
@@ -1,52 +1,81 @@
-import { Request, Response } from "express";
-import { PATH_NAMES } from "../../app.constants";
-import { ExpressRouteFunc } from "../../types";
-import { updateTermsAndCondsService } from "./updated-terms-conds-service";
-import {
-  ClientInfoResponse,
-  UpdateTermsAndCondsServiceInterface,
-} from "./types";
+import {Request, Response} from "express";
+import {PATH_NAMES} from "../../app.constants";
+import {ExpressRouteFunc} from "../../types";
+import {updateTermsAndCondsService} from "./updated-terms-conds-service";
+import {ClientInfoResponse, UpdateTermsAndCondsServiceInterface,} from "./types";
 
 export function updatedTermsCondsGet(
-  service: UpdateTermsAndCondsServiceInterface = updateTermsAndCondsService()
+    service: UpdateTermsAndCondsServiceInterface = updateTermsAndCondsService()
 ): ExpressRouteFunc {
-  return async function (req: Request, res: Response) {
-    const sessionId = res.locals.sessionId;
-    const clientSessionId = res.locals.clientSessionId;
-    const clientInfo: ClientInfoResponse = await service.clientInfo(
-      sessionId,
-      clientSessionId
-    );
-    req.session.redirectUri = clientInfo.redirectUri;
+    return async function (req: Request, res: Response) {
+        const clientInfo = await callClientInfo(req, res, service);
+        req.session.redirectUri = clientInfo.redirectUri;
+        req.session.state = clientInfo.state;
+
+        res.render("updated-terms-conds/index.njk");
+    };
+}
+
+export function updatedTermsCondsMandatoryGet(
```

Comment:
is there a reason for having 2 different routes for each page could it not be done under one? i would speak to suly about this as the page urls are usually defined in figma and eventually will be tracked by GA

Reply:
As discussed, Suly & Kim have confirmed that they want 3 separate URL paths for GA tracking.